src/tests.rs

```diff
@@ -1,2 +1,229 @@
+        assert_eq!(expected_response, format!("{:?}", res));
+
+        let expected_messages = "[SubMsg { id: 0, msg: Bank(Send { to_address: \"owner1\", amount: [Coin { denom: \"earth\", amount: Uint128(450) }] }), gas_limit: None, reply_on: Never }, SubMsg { id: 0, msg: Bank(Send { to_address: \"owner2\", amount: [Coin { denom: \"earth\", amount: Uint128(550) }] }), gas_limit: None, reply_on: Never }]";
+        assert_eq!(expected_messages, format!("{:?}", res.messages));
+
+        assert_eq!(1, res.attributes.len());
+        assert_eq!(2, res.messages.len());
+        assert_eq!(0, res.events.len());
+
+        // Owner2 disburse
+        let msg = ExecuteMsg::Disburse {};
+        let info = mock_info("owner2", &coins(1000, "earth"));
+        let res = execute(deps.as_mut(), env.clone(), info.clone(), msg).unwrap();
+        // println!("ENTIRE RESPONSE: {:?}", res);
+        // println!("MESSAGES: {:?}", res.messages);
+
+        let expected_response = "Response { messages: [SubMsg { id: 0, msg: Bank(Send { to_address: \"owner1\", amount: [Coin { denom: \"earth\", amount: Uint128(450) }] }), gas_limit: None, reply_on: Never }, SubMsg { id: 0, msg: Bank(Send { to_address: \"owner2\", amount: [Coin { denom: \"earth\", amount: Uint128(550) }] }), gas_limit: None, reply_on: Never }], attributes: [Attribute { key: \"disbursed_by\", value: \"owner2\" }], events: [], data: None }";
+        assert_eq!(expected_response, format!("{:?}", res));
+
+        let expected_messages = "[SubMsg { id: 0, msg: Bank(Send { to_address: \"owner1\", amount: [Coin { denom: \"earth\", amount: Uint128(450) }] }), gas_limit: None, reply_on: Never }, SubMsg { id: 0, msg: Bank(Send { to_address: \"owner2\", amount: [Coin { denom: \"earth\", amount: Uint128(550) }] }), gas_limit: None, reply_on: Never }]";
+        assert_eq!(expected_messages, format!("{:?}", res.messages));
+
+        assert_eq!(1, res.attributes.len());
+        assert_eq!(2, res.messages.len());
+        assert_eq!(0, res.events.len());
+    }
+
+    #[test]
+    fn test_instantiate_total_ownership_error() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info("creator", &[]);
+        let msg = InstantiateMsg {
+            owners: vec![
+                Owner {
+                    address: Addr::unchecked("owner1"),
+                    ownership: 40,
+                },
+                Owner {
+                    address: Addr::unchecked("owner2"),
+                    ownership: 50,
+                },
+            ],
+        };
+
+        // Check that the `Total Ownership must equal 100%` error is thrown when the total ownership is greater than 100
+        let res = instantiate(deps.as_mut(), env, info.clone(), msg);
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap(),
+            cosmwasm_std::StdError::generic_err("Total Ownership must equal 100%."),
```

Comment:
Why you create custom errors instead of generic_err with text.
If in the future you will want to change the text? you will need to change all the tests to be the new text.
use custom errors.